buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -189,7 +194,7 @@ final class AdaptivePoolingAllocator {
         largeBufferMagazineGroup = new MagazineGroup(
                 this, chunkAllocator, new HistogramChunkControllerFactory(true), false);
 
-        threadLocalGroup = new FastThreadLocal<MagazineGroup[]>() {
+        threadLocalGroup = IS_LOW_MEM ? null : new FastThreadLocal<MagazineGroup[]>() {
```

Comment:
Especially with the incoming changes in adaptive, the gap between thread local allocations and shared will be wider: I would still provide a perk to let users force using fast thread local ones. 
512 MB is not that small, depending what the application is doing

Reply:
Ok, added a system property.